soc/silabs/silabs_s2/Kconfig

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2025 Silicon Laboratories Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_SILABS_S2
+if MBEDTLS
+
+config PSA_CRYPTO_SW_ONLY
+	bool "PSA Crypto software implementation only"
+
+config PSA_CRYPTO_DRIVER_SILABS_HSE
+	bool "PSA Crypto Driver for Secure Engine"
+	default y if (SOC_SERIES_EFR32MG21 || SOC_SERIES_EFR32MG24 || SOC_SERIES_EFR32BG29 || \
+		      SOC_SERIES_EFR32MG29) && !PSA_CRYPTO_SW_ONLY
+
+config PSA_CRYPTO_DRIVER_SILABS_VSE
+	bool "PSA Crypto Driver for Virtual Secure Engine"
+	# default y if (SOC_SERIES_EFR32BG22 || SOC_SERIES_EFR32BG27)
+
+config MBEDTLS_USER_CONFIG_ENABLE
+	default y if (PSA_CRYPTO_DRIVER_SILABS_HSE || PSA_CRYPTO_DRIVER_SILABS_VSE)
+
+config MBEDTLS_USER_CONFIG_FILE
+	default "sl_mbedtls_config_zephyr.h" \
+		if (PSA_CRYPTO_DRIVER_SILABS_HSE || PSA_CRYPTO_DRIVER_SILABS_VSE)
```

Comment:
Any way to merge `MBEDTLS_USER_CONFIG_ENABLE` and `MBEDTLS_USER_CONFIG_FILE`?

Reply:
These options are defined by upstream in https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/mbedtls/Kconfig.tls-generic#L425-L436
So no, not unless we improve it there. In which case we should probably add acceleration-specific config hooks instead of consuming the user hook. I did it this way in order to be able to do it without changing the upstream.